src/FakeItEasy/ReturnValueConfigurationExtensions.cs

```diff
@@ -324,13 +328,8 @@ public static void ReturnsNextFromSequence<T>(this IReturnValueConfiguration<T>
         [SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures", Justification = "Necessary to support special handling of Task<T> return values.")]
         public static void ReturnsNextFromSequence<T>(this IReturnValueConfiguration<Task<T>> configuration, params T[] values)
         {
-            Guard.AgainstNull(configuration, nameof(configuration));
-
-            var taskValues = values.Select(value => TaskHelper.FromResult(value));
-
-            var queue = new Queue<Task<T>>(taskValues);
-
-            configuration.ReturnsLazily(x => queue.Dequeue()).NumberOfTimes(queue.Count);
+            var taskValues = values.Select(TaskHelper.FromResult).ToArray();
+            configuration.ReturnsNextFromSequence(taskValues);

```

Comment:
Shouldn't there be a null guard for `values`? It will never be null with the "natural" params usage, but it's still possible to pass null explicitly (not really in scope for this PR, though; the problem was already there)